src/libs/antares/study/area/sets.cpp

```diff
@@ -400,64 +367,103 @@ void Sets<T>::rebuildIndexes()
     }
 }
 
-template<class T>
-template<class StringT>
-inline bool Sets<T>::hasOutput(const StringT& s) const
+bool Sets::hasOutput(const uint index) const
 {
-    // Assert, if a C* container can not be found at compile time
-    static_assert(Yuni::Traits::CString<StringT>::valid);
+    return hasOutput(IDType(pNameByIndex[index]));
+}
 
-    typename MapOptions::const_iterator i = pOptions.find(s);
-    return (i != pOptions.end()) ? i->second.output : false;
+Sets::IDType Sets::caption(const uint i) const
+{
+    return caption(IDType(pNameByIndex[i]));
 }
 
-template<class T>
-inline bool Sets<T>::hasOutput(const uint index) const
+uint Sets::resultSize(const uint index) const
 {
-    return hasOutput(IDType(pNameByIndex[index]));
+    return resultSize(IDType(pNameByIndex[index]));
 }
 
-template<class T>
-template<class StringT>
-inline uint Sets<T>::resultSize(const StringT& s) const
+void Sets::dumpToLogs() const
 {
-    // Assert, if a C* container can not be found at compile time
-    static_assert(Yuni::Traits::CString<StringT>::valid);
+    using namespace Yuni;
+    const typename MapType::const_iterator end = pMap.end();
+    for (typename MapType::const_iterator i = pMap.begin(); i != end; ++i)
```

Comment:
```suggestion
    for (const auto& [name, content] : pMap)
```

Reply:
I'll do that, no problem, but we should keep in mind that any **std::map** used here (for instance **pMap**) should not exist : please see PR presentation comment.
**std::map** and  **std::vector** are used because of a chaotic architecture of class **Sets**.

Reply:
Done